tests/tests_pytorch/loops/test_fetchers.py

```diff
@@ -294,7 +301,6 @@ def training_step(self, dataloader_iter: Iterator) -> STEP_OUTPUT:
         self.optimizers().zero_grad()
 
         self.batch_i_handle = batch_ip1_handle
-        self.num_batches_processed += 1
```

Comment:
This was wrong because it doesn't take into account that the first `training_step` calls `next()` twice